controllers/operandbindinfo/operandbindinfo_controller.go

```diff
@@ -521,7 +534,103 @@ func (r *Reconciler) copyRoute(ctx context.Context, route operatorv1alpha1.Route
 	return false, nil
 }
 
-// sanitizedOdlmRouteData takes a map, i.e. ODLM's Route.Data, and an OCP Route.Spec,
+// copyRoute reads the data map and copies K8s Service data as specified by the
+// field path in the data map values
+func (r *Reconciler) copyService(ctx context.Context, service operatorv1alpha1.ServiceData, targetName, sourceNs, targetNs, key string,
+	bindInfoInstance *operatorv1alpha1.OperandBindInfo, requestInstance *operatorv1alpha1.OperandRequest) (requeue bool, err error) {
+	if service.Name == "" || sourceNs == "" || targetNs == "" {
+		return false, nil
+	}
+
+	if service.Name == targetName && sourceNs == targetNs {
+		return false, nil
+	}
+
+	if targetName == "" {
+		if publicPrefix.MatchString(key) {
+			targetName = bindInfoInstance.Name + "-" + service.Name
```

Comment:
Please add the resource type `service` in the targetName, so that it's clear that ConfigMap represents a Service vs Route, vs anything else